src/components/GraphiQL.js

```diff
@@ -478,9 +478,14 @@ export class GraphiQL extends React.Component {
     let jsonVariables = null;
 
     try {
-      jsonVariables = JSON.parse(variables);
+      jsonVariables =
+        variables && variables.trim() !== '' ? JSON.parse(variables) : null;
     } catch (error) {
-      jsonVariables = null;
+      throw new Error(`Variables are invalid JSON: ${error.message}.`);
```

Comment:
I think I would prefer to keep it as a simple string for these 2 reasons:

* As far as I can tell, other parts of GraphiQL also follow this pattern and render errors as a string (`"Fetcher did not return Promise or Observable."` is an example of it). But maybe I missed something?
*  Rendering this error as a JSON may case a confusion because it looks exactly like a server response, even though no request was sent to the server in case of malformed variables. 

WDYT?